pkg/logql/engine.go

```diff
@@ -67,6 +68,11 @@ type EngineOpts struct {
 	MaxLookBackPeriod time.Duration `yaml:"max_look_back_period"`
 }
 
+func(opts *EngineOpts) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.DurationVar(&opts.Timeout, prefix+".engine.timeout", 3*time.Minute, "Timeout for query execution")
+	f.DurationVar(&opts.MaxLookBackPeriod, prefix+".engine.max-lookback-period", 30*time.Second, "MaxLookBackPeriod is the maximum amount of time to look back for log lines. Used only for instant log queries")
```

Comment:
This default is probably too short for a default, what is the behavior now, do we just support infinite lookback if you don't define this?

Reply:
This is only use for instant log queries. I was already the default but we could change it to 5m if you want.

Reply:
5m seems more reasonable to me as a default I think 👍